### ISSUE TITLE
Invoke sparql from live code block

### ISSUE THREAD

**Issue by dvargasfuertes** [dvargas92495/roamjs-developer#6]

**Comment by kvistgaard** [comment]
See a full description [here](https://roamresearch.com/#/app/nodebook/page/DWJVWWI3p)

**Comment by mdroidian** [comment]
Ah, so would you want something like a button on the Sparql code block?  And the results to be outputted as children of the code block?

![image](https://github.com/RoamJS/developer/assets/3792666/d4d343af-6a8c-4460-b0a0-250b108911bc)

**Comment by kvistgaard** [comment]
Not really. And better not. Running it from the command pallette should be
fine and whoever wants will make a custom hotkey.

**Comment by mdroidian** [comment]
Alright.  So I added a new command `Run SPARQL Query: Custom`

This will load to the `Custom Query` page initially and grab the last edited sparql code block from the current page.

What left would then be:

- [ ] define endpoint / output format
- [ ] import endpoint / output format with command
- [ ] support block refs
- [ ] possibly add a command that does all of the above and just runs the query (skip the dialog preview)

EDIT: added tasks
- [ ] run the command when the cursor is on the query block or its parent?
- [ ] add shortcut to block menu to run the query

Hopefully this will be helpful in the interim. 

https://www.loom.com/share/5ad18cdec7164697893e3c698acf4131

**Comment by kvistgaard** [comment]
Wouldn't it be easier to run the command when the cursor is on the query block or its parent?

In addition, it would be nice to be able to start it from the block menu as is the case with some other extensions.

![image](https://github.com/RoamJS/developer/assets/7572721/3c35aebf-72eb-494a-89c4-1af52eb73675)

BTW, what do you mean by 

> import endpoint / output format with command

**Comment by mdroidian** [comment]
> Wouldn't it be easier to run the command when the cursor is on the query block or its parent?

Depends on the use case, I guess.  I believe currently it imports at the cursor position, so if one wanted to import at a different block than the sparql code block, that functionality would be lost.  But I'd be happy to add this feature to the list.
 
> In addition, it would be nice to be able to start it from the block menu as is the case with some other extensions.

This can be added to the list as well.

> import endpoint / output format with command

This was referring to grabbing the endpoint from a comment in the sparql query block and updating the endpoint in the dialog
